libedit has rl_callback_read_char() but not rl_on_new_line().

--- origsrc/php-5.3.8/ext/readline/config.m4	2009-12-13 11:06:47.000000000 -0600
+++ src/php-5.3.8/ext/readline/config.m4	2011-11-09 21:21:20.888295700 -0600
@@ -53,13 +53,6 @@ if test "$PHP_READLINE" && test "$PHP_RE
     -L$READLINE_DIR/$PHP_LIBDIR $PHP_READLINE_LIBS
   ])
 
-  PHP_CHECK_LIBRARY(readline, rl_callback_read_char,
-  [
-    AC_DEFINE(HAVE_RL_CALLBACK_READ_CHAR, 1, [ ])
-  ],[],[
-    -L$READLINE_DIR/$PHP_LIBDIR $PHP_READLINE_LIBS
-  ])
-
   AC_DEFINE(HAVE_LIBREADLINE, 1, [ ])
 
 elif test "$PHP_LIBEDIT" != "no"; then
@@ -97,7 +90,12 @@ elif test "$PHP_LIBEDIT" != "no"; then
 fi
 
 if test "$PHP_READLINE" != "no" || test "$PHP_LIBEDIT" != "no"; then
+  save_LIBS="$LIBS"
+  LIBS="-L$READLINE_DIR/$PHP_LIBDIR $READLINE_SHARED_LIBADD $PHP_READLINE_LIBS"
   AC_CHECK_FUNCS([rl_completion_matches])
+  AC_CHECK_FUNCS([rl_callback_read_char])
+  AC_CHECK_FUNCS([rl_on_new_line])
+  LIBS="$save_LIBS"
   PHP_NEW_EXTENSION(readline, readline.c, $ext_shared, cli)
   PHP_SUBST(READLINE_SHARED_LIBADD)
 fi
--- origsrc/php-5.3.8/ext/readline/readline.c	2011-07-28 05:42:45.000000000 -0500
+++ src/php-5.3.8/ext/readline/readline.c	2011-11-09 21:21:21.595336200 -0600
@@ -50,13 +50,15 @@ PHP_FUNCTION(readline_list_history);
 PHP_FUNCTION(readline_read_history);
 PHP_FUNCTION(readline_write_history);
 PHP_FUNCTION(readline_completion_function);
+#ifdef HAVE_RL_ON_NEW_LINE
+PHP_FUNCTION(readline_on_new_line);
+#endif
 
 #if HAVE_RL_CALLBACK_READ_CHAR
 PHP_FUNCTION(readline_callback_handler_install);
 PHP_FUNCTION(readline_callback_read_char);
 PHP_FUNCTION(readline_callback_handler_remove);
 PHP_FUNCTION(readline_redisplay);
-PHP_FUNCTION(readline_on_new_line);
 
 static zval *_prepped_callback = NULL;
 
@@ -118,7 +120,9 @@ ZEND_END_ARG_INFO()
 
 ZEND_BEGIN_ARG_INFO(arginfo_readline_redisplay, 0)
 ZEND_END_ARG_INFO()
+#endif
 
+#ifdef HAVE_RL_ON_NEW_LINE
 ZEND_BEGIN_ARG_INFO(arginfo_readline_on_new_line, 0)
 ZEND_END_ARG_INFO()
 #endif
@@ -141,6 +145,8 @@ static const zend_function_entry php_rea
 	PHP_FE(readline_callback_read_char,			arginfo_readline_callback_read_char)
 	PHP_FE(readline_callback_handler_remove,	arginfo_readline_callback_handler_remove)
 	PHP_FE(readline_redisplay, arginfo_readline_redisplay)
+#endif
+#ifdef HAVE_RL_ON_NEW_LINE
 	PHP_FE(readline_on_new_line, arginfo_readline_on_new_line)
 #endif
 	PHP_FE_END
@@ -604,6 +610,10 @@ PHP_FUNCTION(readline_redisplay)
 }
 /* }}} */
 
+#endif
+
+#ifdef HAVE_RL_ON_NEW_LINE
+
 /* {{{ proto void readline_on_new_line(void)
    Inform readline that the cursor has moved to a new line */
 PHP_FUNCTION(readline_on_new_line)
